OpenRA.Mods.Common/Lint/CheckConflictingMouseBounds.cs

```diff
@@ -0,0 +1,31 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Lint
+{
+	public class CheckConflictingMouseBounds : ILintRulesPass
+	{
+		public void Run(Action<string> emitError, Action<string> emitWarning, Ruleset rules)
+		{
+			foreach (var actorInfo in rules.Actors)
+			{
+				var selectable = actorInfo.Value.TraitInfos<SelectableInfo>().Count();
+				var interactable = actorInfo.Value.TraitInfos<InteractableInfo>().Count();
+				if (selectable > 0 && selectable != interactable)
```

Comment:
Won't this emit a warning even if `interactable == 0`?

Reply:
`SelectableInfo` inherits `InteractableInfo` so are also included in the `interactable` count.  The idea here is to make sure that there are either only `Interactable` traits (`selectable == 0`), or that there are no `Interactable` traits that are not also `Selectable` (`selectable == interactable`).